lib/improver/tests/optical_flow/test_AdvectField.py

```diff
@@ -0,0 +1,285 @@
+                         [0., 1., 2.],
+                         [0., 0., 1.]])
+        self.cube = iris.cube.Cube(
+            data, standard_name='rainfall_rate', units='mm h-1',
+            dim_coords_and_dims=[(self.plugin.y_coord, 0),
+                                 (self.plugin.x_coord, 1)])
+
+        # input time: [datetime.datetime(2018, 2, 20, 4, 0)]
+        self.time_coord = DimCoord(1519099200, standard_name="time",
+                                   units='seconds since 1970-01-01 00:00:00')
+        self.cube.add_aux_coord(self.time_coord)
+
+        self.timestep = datetime.timedelta(seconds=600)
+
+    def test_basic(self):
+        """Test output cube data is as expected"""
+        expected_output = np.array([[0., 0., 0.],
+                                    [0., 0., 0.],
+                                    [0., 2., 3.],
+                                    [0., 1., 2.]])
+        result = self.plugin.process(self.cube, self.timestep)
+        self.assertArrayAlmostEqual(result.data, expected_output)
+
+    def test_background_values(self):
+        """Test output cube data is padded as expected where source grid
+        points are out of bounds"""
+        expected_output = np.array([[-1., -1., -1.],
+                                    [-1., -1., -1.],
+                                    [-1., 2., 3.],
+                                    [-1., 1., 2.]])
+        result = self.plugin.process(self.cube, self.timestep, bgd=-1.)
+        self.assertArrayAlmostEqual(result.data, expected_output)
+
+    def test_time_step(self):
+        """Test outputs are OK for a time step with non-second components"""
+        expected_output = np.zeros(shape=(4, 3))
+        result = self.plugin.process(self.cube, datetime.timedelta(hours=1))
+        self.assertArrayAlmostEqual(result.data, expected_output)
+
+    def test_raises_grid_mismatch_error(self):
+        """Test error is raised if cube grid does not match velocity grids"""
+        x_coord = DimCoord(np.arange(5), 'projection_x_coordinate', units='km')
+        y_coord = DimCoord(np.arange(4), 'projection_y_coordinate', units='km')
+        cube = iris.cube.Cube(np.zeros(shape=(4, 5)),
+                              standard_name='rainfall_rate', units='mm h-1',
+                              dim_coords_and_dims=[(y_coord, 0), (x_coord, 1)])
+        cube.add_aux_coord(self.time_coord)
+
+        msg = "Input data grid does not match advection velocities"
+        with self.assertRaisesRegexp(InvalidCubeError, msg):
+            self.plugin.process(cube, self.timestep)
+
+    def test_validity_time(self):
+        """Test output cube time is correctly updated"""
+        result = self.plugin.process(self.cube, self.timestep)
+        output_cube_time, = \
+            (result.coord("time").units).num2date(result.coord("time").points)
+        self.assertEqual(output_cube_time.year, 2018)
+        self.assertEqual(output_cube_time.month, 02)
```

Comment:
Please remove leading zero from this number. In this format, it can be interpreted as an octal number in some circumstances.